{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "openstack-nova",
                    "vendor": "n/a",
                    "versions": [
                        {
                            "status": "affected",
                            "version": "Affects - Nova: <21.2.3, >=22.0.0 <22.2.3, >=23.0.0 <23.0.3 | Fixed-In 21.2.3, 22.3.0, and 23.1.0"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "A vulnerability was found in openstack-nova's console proxy, noVNC. By crafting a malicious URL, noVNC could be made to redirect to any desired URL."
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-601",
                            "description": "CWE-601 - URL Redirection to Untrusted Site ('Open Redirect')",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2022-03-02T22:08:06",
                "orgId": "53f830b8-0a3f-465b-8143-3b8a9948e749",
                "shortName": "redhat"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://security.openstack.org/ossa/OSSA-2021-002.html"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://bugs.launchpad.net/nova/+bug/1927677"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://www.openwall.com/lists/oss-security/2021/07/29/2"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://bugs.python.org/issue32084"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://opendev.org/openstack/nova/commit/04d48527b62a35d912f93bc75613a6cca606df66"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://opendev.org/openstack/nova/commit/8906552cfc2525a44251d4cf313ece61e57251eb"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1961439"
                }
            ],
            "x_legacyV4Record": {
                "CVE_data_meta": {
                    "ASSIGNER": "secalert@redhat.com",
                    "ID": "CVE-2021-3654",
                    "STATE": "PUBLIC"
                },
                "affects": {
                    "vendor": {
                        "vendor_data": [
                            {
                                "product": {
                                    "product_data": [
                                        {
                                            "product_name": "openstack-nova",
                                            "version": {
                                                "version_data": [
                                                    {
                                                        "version_value": "Affects - Nova: <21.2.3, >=22.0.0 <22.2.3, >=23.0.0 <23.0.3 | Fixed-In 21.2.3, 22.3.0, and 23.1.0"
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                },
                                "vendor_name": "n/a"
                            }
                        ]
                    }
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "eng",
                            "value": "A vulnerability was found in openstack-nova's console proxy, noVNC. By crafting a malicious URL, noVNC could be made to redirect to any desired URL."
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "CWE-601 - URL Redirection to Untrusted Site ('Open Redirect')"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://security.openstack.org/ossa/OSSA-2021-002.html",
                            "refsource": "MISC",
                            "url": "https://security.openstack.org/ossa/OSSA-2021-002.html"
                        },
                        {
                            "name": "https://bugs.launchpad.net/nova/+bug/1927677",
                            "refsource": "MISC",
                            "url": "https://bugs.launchpad.net/nova/+bug/1927677"
                        },
                        {
                            "name": "https://www.openwall.com/lists/oss-security/2021/07/29/2",
                            "refsource": "MISC",
                            "url": "https://www.openwall.com/lists/oss-security/2021/07/29/2"
                        },
                        {
                            "name": "https://bugs.python.org/issue32084",
                            "refsource": "MISC",
                            "url": "https://bugs.python.org/issue32084"
                        },
                        {
                            "name": "https://opendev.org/openstack/nova/commit/04d48527b62a35d912f93bc75613a6cca606df66",
                            "refsource": "MISC",
                            "url": "https://opendev.org/openstack/nova/commit/04d48527b62a35d912f93bc75613a6cca606df66"
                        },
                        {
                            "name": "https://opendev.org/openstack/nova/commit/8906552cfc2525a44251d4cf313ece61e57251eb",
                            "refsource": "MISC",
                            "url": "https://opendev.org/openstack/nova/commit/8906552cfc2525a44251d4cf313ece61e57251eb"
                        },
                        {
                            "name": "https://bugzilla.redhat.com/show_bug.cgi?id=1961439",
                            "refsource": "MISC",
                            "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1961439"
                        }
                    ]
                }
            }
        }
    },
    "cveMetadata": {
        "assignerOrgId": "53f830b8-0a3f-465b-8143-3b8a9948e749",
        "assignerShortName": "redhat",
        "cveId": "CVE-2021-3654",
        "datePublished": "2022-03-02T22:08:06",
        "dateReserved": "2021-07-20T00:00:00",
        "dateUpdated": "2022-03-02T22:08:06",
        "state": "PUBLISHED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}